types/ember/test/ember-module-tests.ts

```diff
@@ -0,0 +1,280 @@
+    },
+    set(_key: string, newVal: string): string {
+        return '';
+    }
+});
+// Ember.ContainerDebugAdapter
+const cda = new Ember.ContainerDebugAdapter(); // $ExpectType ContainerDebugAdapter
+// Ember.Controller
+const con = new Ember.Controller(); // $ExpectType Controller
+// Ember.CoreObject
+const co = new Ember.CoreObject(); // $ExpectType CoreObject
+// Ember.DataAdapter
+const da = new Ember.DataAdapter(); // $ExpectType DataAdapter
+// Ember.Debug
+Ember.Debug.registerDeprecationHandler(() => {});
+Ember.Debug.registerWarnHandler(() => {});
+// Ember.DefaultResolver
+const dr = new Ember.DefaultResolver();
+dr.resolve('route:index');
+// Ember.Engine
+const e1 = new Ember.Engine();
+e1.register('data:foo', {}, { instantiate: false });
+// Ember.EngineInstance
+const ei1 = new Ember.EngineInstance();
+ei1.lookup('data:foo');
+// Ember.Error
+new Ember.Error('Halp!');
+// Ember.Evented
+const oe1 = EmberObject.extend(Evented).create();
+oe1.trigger('foo');
+oe1.on('bar', () => {});
+// Ember.HashLocation
+const hl = new Ember.HashLocation();
+// Ember.Helper
+const h1 = Ember.Helper.extend({
+    compute() {
+        this.recompute();
+        return '';
+    }
+});
+// Ember.HistoryLocation
+const hil = new Ember.HistoryLocation();
+// Ember.Instrumentation
+Ember.Instrumentation.instrument('foo', {}, () => {}, window);
+Ember.Instrumentation.reset();
+// Ember.LinkComponent
+Ember.LinkComponent.create(); // $ExpectType LinkComponent
+// Ember.Mixin
+Ember.Object.extend(Ember.Mixin.create({ foo: 'bar' }), {
+    baz() {
+        this.foo; // $ExpectType string
+    }
+});
+// Ember.MutableArray
+const ma1: Ember.MutableArray<string> = [
+    'money',
+    'in',
+    'the',
+    'bananna',
+    'stand'
+];
+ma1.addObject('!');
+ma1.filterBy('');
+// Ember.MutableEnumerable
+// tslint:disable-next-line:prefer-const
+let me1: Ember.MutableEnumerable<[string]> = null as any;
+me1.compact();
+// Ember.Namespace
+const myNs = Ember.Namespace.extend({});
+// Ember.NativeArray
+const na: Ember.NativeArray<number> = Ember.A([2, 3, 4]);
+na; // $ExpectType NativeArray<number>
+na.clear();
+// Ember.NoneLocation
+new Ember.NoneLocation();
+// Ember.Object
+new Ember.Object();
+// Ember.ObjectProxy
+new Ember.ObjectProxy();
+// Ember.Observable
+Ember.Object.extend(Ember.Observable, {});
+// Ember.PromiseProxyMixin
+Ember.Object.extend(Ember.PromiseProxyMixin, {
+    foo() {
+        this.reason; // $ExpectType any
+        this.isPending; // $ExpectType boolean
+    }
+});
+// Ember.Route
+new Ember.Route();
+// Ember.Router
+new Ember.Router();
+// Ember.Service
+new Ember.Service();
+// Ember.Test
+Ember.Test;
+// Ember.Test.Adapter
+new Ember.Test.Adapter();
+// Ember.Test.QUnitAdapter
+new Ember.Test.QUnitAdapter();
+// Ember.TextArea
+new Ember.TextArea();
+// Ember.TextField
+new Ember.TextField();
+// Ember.Helper
+// helper
+Ember.Helper.helper(([a, b]: [number, number]) => a + b);
+// Ember.String
+Ember.String;
```

Comment:
expect type String

Reply:
this is going to be fail, since `String` is a built-in type. 

Exhaustive string testing is already here

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/ember/test/string.ts